app/src/main/java/ch/epfl/sdp/blindly/chat/ChatActivity.kt

```diff
@@ -66,23 +69,16 @@ class ChatActivity : AppCompatActivity() {
      * @param view the current view
      */
     fun sendButtonActivate(view: View) {
-        findViewById<ImageView>(R.id.sendButton).setOnClickListener {
-            if (findViewById<EditText>(R.id.newMessageText).text.toString().isNotEmpty()) {
-                sendMessage()
-            }
+        if (findViewById<EditText>(R.id.newMessageText).text.toString().isNotEmpty()) {
```

Comment:
Since you removed the .setOnClickListener, is it directly mapped to the button in the .xml?

Reply:
It always has been, however how the listener was implemented deactivated the button before the first click, which was causing troubles with the test. I don't know why it was implemented like that in the first place, maybe @cgntnr you have more insight on this?

Reply:
I think the first click being deactivated is about focusable attribute. In terms of sendButton yes, the xml file already had the onClick attribute which called this function and if the input(newMessageText) is not empty calls the sendMessage to send the text to database and set the view. I didn't see any disadvantage really, less boilerplate code imo but I'm open to discussion/change if you feel otherwise.

Reply:
Yes, but setting the correct click listener after the first click requires to click to send a message on the first time.
Like:
-  click 1:
    - set the click listener to (not empty -> send message)
-  click 2:
    - calls (not empty -> send message)
      - if not empty, send message

Reply:
I think this [answer](https://stackoverflow.com/questions/46289043/android-onclicklistener-not-working-for-first-click-in-recyclerview) could solve the issue.

Reply:
I think I'm not really understanding what your are trying to do. Why set the actual click listener only after the second click? I don't think your code match your comment either, it is not 
>  checks the input EditText and if it's empty, calls the sendMessage.

but rather
>  attach a new click listener so that next time the button is click, it checks the input EditText and if it's empty, calls the sendMessage.

Anyway, do you think that my change breaks something you wanted to achieve, or can I merge ?
Sorry if I'm a bit confused about this part.

Reply:
It calls only if it's not empty! To  my understanding there wasn't a second onListener but maybe I skipped something. Well no worries I think it got a bit confusing, but I tested out your code and seemed working fine so no worries go ahead and merge.